src/ring_theory/ideal/operations.lean

```diff
@@ -1051,6 +1051,41 @@ by { rw [mem_ker, f.map_one], exact one_ne_zero }
 @[simp] lemma ker_coe_equiv (f : R ≃+* S) : ker (f : R →+* S) = ⊥ :=
 by simpa only [←injective_iff_ker_eq_bot] using f.injective
 
+/-- The induced map from the quotient by the kernel to the codomain. -/
+def ker_lift (f : R →+* S) : f.ker.quotient →+* S :=
+ideal.quotient.lift _ f $ λ r, f.mem_ker.mp
+
+@[simp]
+lemma ker_lift_mk (f : R →+* S) (r : R) : ker_lift f (ideal.quotient.mk f.ker r) = f r :=
+ideal.quotient.lift_mk _ _ _
+
+/-- The induced map from the quotient by the kernel is injective. -/
+lemma ker_lift_injective (f : R →+* S) : function.injective (ker_lift f) :=
+assume a b, quotient.induction_on₂' a b $
+  assume a b (h : f a = f b), quotient.sound' $
+show a - b ∈ ker f, by rw [mem_ker, map_sub, h, sub_self]
+
+variable {f}
+
+/-- The first isomorphism theorem for commutative rings, computable version. -/
+def quotient_ker_equiv_of_right_inverse
+  {g : S → R} (hf : function.right_inverse g f) :
+  f.ker.quotient ≃+* S :=
+{ to_fun := ker_lift f,
+  inv_fun := (ideal.quotient.mk f.ker) ∘ g,
+  left_inv := begin
+    rintro ⟨x⟩,
+    apply ker_lift_injective,
+    simp [hf (f x)],
+  end,
+  right_inv := hf,
+  ..ker_lift f}
+
+/-- The first isomorphism theorem for commutative rings. -/
+noncomputable def quotient_ker_equiv_of_surjective (hf : function.surjective f) :
+  f.ker.quotient ≃+* S :=
+quotient_ker_equiv_of_right_inverse (classical.some_spec hf.has_right_inverse)
```

Comment:
Should we add some `simp` lemmas saying `quotient_ker_equiv_of_right_inverse hf x = ker_lift x`, `(quotient_ker_equiv_of_right_inverse hf).symm x = mk (g x)`, etc?